Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -210,6 +212,23 @@ public void Serialize<T>(IIonWriter writer, T item)
             }
 
             Type type = item.GetType();
+
+            if (options.AnnotatedIonSerializers != null)

```

Comment:
Do these `AnnotatedIonSerializers` happen with primitive types or just objects?
If the latter then this logic should happen within the object logic.

Reply:
Added as a part of the rework.